README.md

```diff
@@ -30,6 +31,80 @@ load("@io_bazel_rules_k8s//k8s:k8s.bzl", "k8s_repositories")
 k8s_repositories()
 ```
 
-## SkyDoc
+## Kubernetes Authentication
+
+As is somewhat standard for Bazel, the expectation is that the
+`kubectl` toolchain is preconfigured to authenticate with any clusters
+you might interact with.
+
+For more information on how to configure `kubectl` authentication, see the
+Kubernetes [documentation](https://kubernetes.io/docs/admin/authentication/).
+
+### Container Engine Authentication
+
+For Google Container Engine (GKE), the `gcloud` CLI provides a [simple
+command](https://cloud.google.com/sdk/gcloud/reference/container/clusters/get-credentials)
+for setting up authentication:
+```shell
+gcloud container clusters get-credentials <CLUSTER NAME>
+```
+
+## Examples
+
+### Basic "deployment" objects
+
+```python
+load("@io_bazel_rules_k8s//k8s:object.bzl", "k8s_object")
+
+k8s_object(
+  name = "dev",
+  kind = "deployment",
```

Comment:
Do you need a `version` field in k8s object at all? The docs still say to use apps/v1beta1 for Deployment: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/

Reply:
Ah I see, the user is expected to set that themselves in the object. What is the "kind" used for then? It doesn't look like anything right now.

Reply:
Nothing yet, it is ultimately used for `kubectl` subcommands.

Reply:
Ah, it just seems a little weird to have it here and in the file itself. Is there anything coming to automatically extract it from the file? What if it doesn't match?

Reply:
IDK if we'll extract it, but we may add validation that the object we're dealing with matches what's in the file (if such a thing is possible).  Ultimately, with CRDs, there doesn't need to be any correlation between the `kind` in the resource and the command-line keyword.